### ISSUE TITLE
Allow comparing of value-only struct/tuple types

### ISSUE THREAD

**Issue by Marenz** [ethereum/solidity#9598]
It should be possible to compare structs and tuples that consist only of simple value types.
If every individual type of the struct/tuple supports `==` then the struct/tuple should also support the operation.

Example:

```
   struct S {
          int x;
          int y;
          int z;
      }
      function f() public {

          S memory s;
          S memory x;
          s == s;
          s == x;
      }
```

```
int x; int y; int z;
(x, y, z) == (int(-1), int(-1), int(-1));
(x, y, z) == (z, y, x);
```

**Comment by leonardoalt** [comment]
Tuple comparison was also requested in https://github.com/ethereum/solidity/issues/8919, could probably be done at the same time

**Comment by cameel** [comment]
We discussed it on the call today:
1. It's very low priority until there's actual demand for it but we wouldn't mind having this feature. Moved to Icebox for now. It's tagged as `good first issue` so anyone is welcome to try, but it's not necessarily an easy issue for someone new (hence `difficulty: medium` tag).
2. An important thing would be for the error message to clearly indicate which members prevent the struct/tuple from being comparable. @Marenz suggested that there should even be a way to tell the compiler to skip those members, though I think it would be hard to find readable and unobtrusive syntax for this.
